tutorial/markdown/nodejs/quickstart-typescript/quickstart-typescript.md

```diff
@@ -0,0 +1,486 @@
+
+Any dependencies will be installed by running the npm install command from the root directory of the project.
+
+```shell
+npm install
+```
+
+### Setup Database Configuration
+
+To know more about connecting to your Capella cluster, please follow the [instructions](https://docs.couchbase.com/cloud/get-started/connect.html).
+
+Specifically, you need to do the following:
+
+- Create the [database credentials](https://docs.couchbase.com/cloud/clusters/manage-database-users.html) to access the travel-sample bucket (Read and Write) used in the application.
+- [Allow access](https://docs.couchbase.com/cloud/clusters/allow-ip-address.html) to the Cluster from the IP on which the application is running.
+
+All configuration for communication with the database is read from the environment variables. We have provided a convenience feature in this quickstart to read the environment variables from a local file, `dev.env` in the `config` folder.
+
+Create a copy of `config/dev.env.example` file and rename it to `config/dev.env` and add the values for the Couchbase connection.
+
+```sh
+CONNECTION_STRING=<connection_string>
+USERNAME=<user_with_read_write_permission_to_travel-sample_bucket>
+PASSWORD=<password_for_user>
+```
+
+> Note: The connection string expects the `couchbases://` or `couchbase://` part.
+
+## Running the Application
+
+### Directly on Local Machine
+
+At this point, we have installed the dependencies, loaded the travel-sample data and configured the application with the credentials. The application is now ready and you can run it.
+
+The application will run on the port specified by the `APP_PORT` variable in your configuration environment variable file on your local machine (eg: http://localhost:3000). You will find the Swagger documentation of the API which you can use to try the API endpoints.
+
+```sh
+# Execute this command in the project's root directory
+npm start
+```
+
+### Docker
+
+If you prefer to run this quick start using Docker, we have provided the Dockerfile which you can use to build the image and run the API as a container.
+
+- Build the Docker image
+
+```sh
+# Execute this command in the project's root directory
+docker build -t couchbase-typescriptjs-quickstart .
+```
+
+- Run the Docker image
+
+```sh
+# Execute this command in the project's root directory. Replace APP_PORT with the value of your APP_PORT variable.
+docker run -it --env-file config/dev.env -p ${APP_PORT}:${APP_PORT} couchbase-typescript-express-js-quickstart
+# For eg: docker run -it --env-file config/dev.env -p 3000:3000 couchbase-typescript-express-js-quickstart
+```
+
+> Note: The `config/dev.env` file has the connection information to connect to your Capella cluster. With the `--env-file`, docker will inject those environment variables to the container.
+
+Once the app is up and running, you can launch your browser and go to the [Swagger documentation](eg: https://localhost:3000/ if APP_PORT is set to 3000) to test the APIs.
```

Comment:
The link would break here. You can just link to localhost:3000. You can mention about the PORT just once with the configuration that this tutorial assumes that you are setting APP_PORT to 3000. If it is set differently, the port would change.

Reply:
Got it, Updated.